containers/rust-postgres/.devcontainer/docker-compose.yml

```
@@ -0,0 +1,54 @@
+version: '3.8'
+
+volumes:
+  postgres-data:
+
+services:
+  app:
+    container_name: rust
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        # Use the VARIANT arg to pick a Debian OS version: buster, bullseye
+        # Use bullseye when on local on arm64/Apple Silicon.
+        VARIANT:  buster
+        # Options
+        NODE_VERSION: "lts/*"
+    env_file:
+        # Ensure that the variables in .env match the same variables in devcontainer.json
+        - .env
+    # Security Opt and cap_add.
+    # See `runArgs`: https://github.com/Microsoft/vscode-docs/blob/main/docs/remote/devcontainerjson-reference.md
+    security_opt:
+      - seccomp:unconfined
+    cap_add:
+      - SYS_PTRACE
+
+    volumes:
+      - ..:/workspace:cached
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+
+    # Uncomment the next line to use a non-root user for all processes.
+    # user: vscode
```

Let's reverse this so we use the least privileged access unless needed. 

```suggestion
    # Comment out the next line to use a root user for all processes.
    user: vscode
```